<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>A Revival of the ClaNC Algorithm â€¢ reclanc</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="A Revival of the ClaNC Algorithm">
<meta name="description" content="Classification of microarrays to nearest centroids (ClaNC) &lt;doi:10.1093/bioinformatics/bti756&gt; selects optimal genes for centroids, similar to Prediction Analysis for Microarrays (PAM) but using fewer corrective factors, resulting in greater sensitivity and accuracy. Unfortunately, the original source of ClaNC can no longer be found. reclanc reimplements this algorithm, with the the additional benefit of increased interoperability with standard data structures and modeling ecosystems.">
<meta property="og:description" content="Classification of microarrays to nearest centroids (ClaNC) &lt;doi:10.1093/bioinformatics/bti756&gt; selects optimal genes for centroids, similar to Prediction Analysis for Microarrays (PAM) but using fewer corrective factors, resulting in greater sensitivity and accuracy. Unfortunately, the original source of ClaNC can no longer be found. reclanc reimplements this algorithm, with the the additional benefit of increased interoperability with standard data structures and modeling ecosystems.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">reclanc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/case-study.html">case-study</a></li>
    <li><a class="dropdown-item" href="articles/using-reclanc.html">using-reclanc</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/KaiAragaki/reclanc/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="reclanc">reclanc<a class="anchor" aria-label="anchor" href="#reclanc"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>reclanc is a revival of <a href="https://doi.org/10.1093/bioinformatics/bti681" class="external-link">ClaNC (Classification of microarrays to nearest centroids), by Alan R. Dabney</a>.</p>
<p>Since the source has been lost (at least to my knowledge), the code comes from <a href="https://github.com/naikai/sake/blob/master/R/clanc.R" class="external-link">here</a> with heavy modification.</p>
<p>reclanc is a nearest-centroid classifier for expression data. It tends to be a little more sensitive and accurate than similar models like PAM.</p>
<p>Besides its mere existence, reclanc differs slightly from the original ClaNC package in a few ways:</p>
<ul>
<li>reclanc supports a wide variety of inputs (<code>data.frame</code>, <code>matrix</code>, <code>formula</code>, <code>recipe</code>, <code>ExpressionSet</code>, and <code>SummarizedExperiment</code>)</li>
<li>reclanc plays nicely with <a href="https://www.tidymodels.org/" class="external-link">tidymodels</a>, and offloads things like making folds to <code>rsample</code> and tuning to <code>tune</code> (see <a href="https://kaiaragaki.github.io/reclanc/articles/case-study.html">this vignette</a> for how to leverage tidymodels with reclanc).</li>
<li>Provides a prediction method based on correlation, rather than distance - useful for predicting classes from data from different sequencing modalities</li>
</ul>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of reclanc like so:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("pak")</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pak.html" class="external-link">pak</a></span><span class="op">(</span><span class="st">"KaiAragaki/reclanc"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="how-to-use-it">How to use it<a class="anchor" aria-label="anchor" href="#how-to-use-it"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KaiAragaki/reclanc" class="external-link">reclanc</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">synthetic_expression</span>, <span class="va">head</span><span class="op">)</span> <span class="co"># dummy data</span></span>
<span><span class="co">#&gt; $expression</span></span>
<span><span class="co">#&gt;        sample1  sample2  sample3  sample4  sample5  sample6  sample7  sample8</span></span>
<span><span class="co">#&gt; gene1 8.097529 7.119188 7.304400 7.554689 7.953206 7.714925 7.512700 8.597547</span></span>
<span><span class="co">#&gt; gene2 8.641837 9.400416 8.500865 8.878687 8.318438 8.728683 7.812591 7.638167</span></span>
<span><span class="co">#&gt; gene3 3.436236 4.317915 3.435193 3.515755 3.024976 4.762209 5.048956 2.006646</span></span>
<span><span class="co">#&gt; gene4 4.368008 5.212750 4.618249 4.201365 3.195294 4.707750 5.126769 6.178658</span></span>
<span><span class="co">#&gt; gene5 2.423974 3.563816 4.062362 2.163278 2.021435 2.813873 0.000000 4.652358</span></span>
<span><span class="co">#&gt; gene6 5.371205 5.919809 4.366915 4.805534 4.834856 5.622157 3.883531 3.593082</span></span>
<span><span class="co">#&gt;        sample9 sample10 sample11 sample12</span></span>
<span><span class="co">#&gt; gene1 6.475641 7.648858 8.637526 7.345038</span></span>
<span><span class="co">#&gt; gene2 8.110285 7.906104 7.424728 7.927039</span></span>
<span><span class="co">#&gt; gene3 2.739211 3.111668 3.161077 4.306611</span></span>
<span><span class="co">#&gt; gene4 5.170265 4.259578 5.872855 6.159023</span></span>
<span><span class="co">#&gt; gene5 1.532242 3.399823 3.691250 1.932937</span></span>
<span><span class="co">#&gt; gene6 4.246205 4.637316 3.575837 2.730452</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $classes</span></span>
<span><span class="co">#&gt; [1] A A A A A A</span></span>
<span><span class="co">#&gt; Levels: A B</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">centroids</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/clanc.html">clanc</a></span><span class="op">(</span></span>
<span>  <span class="va">synthetic_expression</span><span class="op">$</span><span class="va">expression</span>,</span>
<span>  classes <span class="op">=</span> <span class="va">synthetic_expression</span><span class="op">$</span><span class="va">classes</span>,</span>
<span>  active <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span>
<span><span class="va">centroids</span></span>
<span><span class="co">#&gt; &lt;clanc&gt; </span></span>
<span><span class="co">#&gt; $centroids</span></span>
<span><span class="co">#&gt;    class   gene expression pooled_sd active prior</span></span>
<span><span class="co">#&gt; 1      A gene13   8.936462 0.3418472      5   0.5</span></span>
<span><span class="co">#&gt; 2      A gene21   7.379940 0.5279636      5   0.5</span></span>
<span><span class="co">#&gt; 3      A  gene2   8.744821 0.3147537      5   0.5</span></span>
<span><span class="co">#&gt; 4      A gene74   4.028507 0.4940783      5   0.5</span></span>
<span><span class="co">#&gt; 5      A gene41   4.328516 0.6317005      5   0.5</span></span>
<span><span class="co">#&gt; 6      A gene66   6.124761 0.5883218      5   0.5</span></span>
<span><span class="co">#&gt; 7      A gene24   4.307301 0.7214700      5   0.5</span></span>
<span><span class="co">#&gt; 8      A gene95   6.288173 0.4462475      5   0.5</span></span>
<span><span class="co">#&gt; 9      A gene94   7.777318 0.5375914      5   0.5</span></span>
<span><span class="co">#&gt; 10     A gene52   3.743798 0.5173769      5   0.5</span></span>
<span><span class="co">#&gt; 11     B gene13   9.938137 0.3418472      5   0.5</span></span>
<span><span class="co">#&gt; 12     B  gene2   8.273987 0.3147537      5   0.5</span></span>
<span><span class="co">#&gt; 13     B gene21   6.584681 0.5279636      5   0.5</span></span>
<span><span class="co">#&gt; 14     B gene41   5.518354 0.6317005      5   0.5</span></span>
<span><span class="co">#&gt; 15     B gene74   3.226598 0.4940783      5   0.5</span></span>
<span><span class="co">#&gt; 16     B gene24   3.370467 0.7214700      5   0.5</span></span>
<span><span class="co">#&gt; 17     B gene66   7.008174 0.5883218      5   0.5</span></span>
<span><span class="co">#&gt; 18     B gene94   8.422255 0.5375914      5   0.5</span></span>
<span><span class="co">#&gt; 19     B gene95   5.703161 0.4462475      5   0.5</span></span>
<span><span class="co">#&gt; 20     B gene52   2.438579 0.5173769      5   0.5</span></span></code></pre></div>
<p>For information on basic usage, see <a href="https://kaiaragaki.github.io/reclanc/articles/using-reclanc.html">this</a> vignette. For a case study, as well as how to optimize the <code>active</code> parameter, see <a href="https://kaiaragaki.github.io/reclanc/articles/case-study.html">this</a> vignette.</p>
</div>
<div class="section level2">
<h2 id="how-it-works">How it works<a class="anchor" aria-label="anchor" href="#how-it-works"></a>
</h2>
<p>You can find a gentle introduction to how reclanc works <a href="https://kai.rbind.io/posts/projects-reclanc/" class="external-link">here</a> and a more in-depth and statistically rigorous definition of how this algorithm works in the <a href="https://doi.org/10.1093/bioinformatics/bti681" class="external-link">original paper</a>.</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Citation for the original ClaNC paper:</p>
<p>Alan R. Dabney, Classification of microarrays to nearest centroids, Bioinformatics, Volume 21, Issue 22, November 2005, Pages 4148â€“4154, <a href="https://doi.org/10.1093/bioinformatics/bti681" class="external-link uri">https://doi.org/10.1093/bioinformatics/bti681</a></p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/KaiAragaki/reclanc/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/KaiAragaki/reclanc/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing reclanc</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Kai Aragaki <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-9458-0426" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Alan Dabney <br><small class="roles"> Author, copyright holder </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Kai Aragaki, Alan Dabney.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
