<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Calculate centroids from expression data with ClaNC — clanc • reclanc</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Calculate centroids from expression data with ClaNC — clanc"><meta name="description" content="Calculate centroids from expression data with ClaNC"><meta property="og:description" content="Calculate centroids from expression data with ClaNC"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">reclanc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/case-study.html">case-study</a></li>
    <li><a class="dropdown-item" href="../articles/using-reclanc.html">using-reclanc</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/KaiAragaki/reclanc/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Calculate centroids from expression data with ClaNC</h1>
      <small class="dont-index">Source: <a href="https://github.com/KaiAragaki/reclanc/blob/master/R/clanc-fit.R" class="external-link"><code>R/clanc-fit.R</code></a></small>
      <div class="d-none name"><code>clanc.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Calculate centroids from expression data with ClaNC</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">clanc</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Default S3 method</span></span>
<span><span class="fu">clanc</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for class 'data.frame'</span></span>
<span><span class="fu">clanc</span><span class="op">(</span><span class="va">x</span>, <span class="va">classes</span>, <span class="va">active</span>, priors <span class="op">=</span> <span class="st">"equal"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for class 'matrix'</span></span>
<span><span class="fu">clanc</span><span class="op">(</span><span class="va">x</span>, <span class="va">classes</span>, <span class="va">active</span>, priors <span class="op">=</span> <span class="st">"equal"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">clanc</span><span class="op">(</span><span class="va">x</span>, <span class="va">classes</span>, <span class="va">active</span>, priors <span class="op">=</span> <span class="st">"equal"</span>, assay <span class="op">=</span> <span class="fl">1</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for class 'ExpressionSet'</span></span>
<span><span class="fu">clanc</span><span class="op">(</span><span class="va">x</span>, <span class="va">classes</span>, <span class="va">active</span>, priors <span class="op">=</span> <span class="st">"equal"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for class 'formula'</span></span>
<span><span class="fu">clanc</span><span class="op">(</span><span class="va">formula</span>, <span class="va">data</span>, <span class="va">active</span>, priors <span class="op">=</span> <span class="st">"equal"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for class 'recipe'</span></span>
<span><span class="fu">clanc</span><span class="op">(</span><span class="va">x</span>, <span class="va">data</span>, <span class="va">active</span>, priors <span class="op">=</span> <span class="st">"equal"</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>Depending on the context:</p><ul><li><p>A <strong>data frame</strong> of expression.</p></li>
<li><p>A <strong>matrix</strong> of expression.</p></li>
<li><p>A <strong>recipe</strong> specifying a set of preprocessing steps
created from <code><a href="https://recipes.tidymodels.org/reference/recipe.html" class="external-link">recipes::recipe()</a></code>.</p></li>
<li><p>An <strong>ExpressionSet</strong>.</p></li>
<li><p>A <strong>SummarizedExperiment</strong> with <code>assay</code> containing expression.</p></li>
</ul><p>Expression should be library-size corrected, but not scaled.</p>
<p>If supplying a <strong>data frame</strong>, <strong>matrix</strong>, <strong>ExpressionSet</strong>,
<strong>SummarizedExperiment</strong>, the rows should represent genes, and the columns
should represent samples (as is standard for expression data). The column
names should be sample IDs, while the row names should be gene IDs.</p>
<p>If a <strong>recipe</strong> is provided, the data should have genes as
columns (to match the formula provided to the recipe.)</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Not currently used, but required for extensibility.</p></dd>


<dt id="arg-classes">classes<a class="anchor" aria-label="anchor" href="#arg-classes"></a></dt>
<dd><p>When <code>x</code> is a <strong>data frame</strong> or <strong>matrix</strong>, <code>class</code> contains
class labels with the form of either:</p><ul><li><p>A <strong>data frame</strong> with 1 factor column</p></li>
<li><p>A factor <strong>vector</strong>.</p></li>
</ul><p>When <code>x</code> is an <strong>ExpressionSet</strong> or <strong>SummarizedExperiment</strong>, <code>class</code> is the
name of the column in <code>pData(x)</code> or <code>colData(x)</code> that contains classes as a
factor.</p></dd>


<dt id="arg-active">active<a class="anchor" aria-label="anchor" href="#arg-active"></a></dt>
<dd><p>Either a single number or a numeric vector equal to the length
of the number of unique class labels. Represents the number class-specific
genes that should be selected for a centroid. Note that different numbers
of genes can be selected for each class. See details.</p>
<p>When <code>x</code> is an <strong>ExpressionSet</strong> or <strong>SummarizedExperiment</strong>, <code>active</code> can
additionally by the name of the column in <code>pData(x)</code> or <code>colData(x)</code> that
contains the numeric vector</p></dd>


<dt id="arg-priors">priors<a class="anchor" aria-label="anchor" href="#arg-priors"></a></dt>
<dd><p>Can take a variety of values:</p><ul><li><p>"equal" - each class has an equal prior</p></li>
<li><p>"class" - each class has a prior equal to its frequency in the training
set</p></li>
<li><p>A numeric vector with length equal to number of classes</p></li>
</ul><p>When <code>x</code> is an <strong>ExpressionSet</strong> or <strong>SummarizedExperiment</strong>, <code>active</code> can
additionally by the name of the column in <code>pData(x)</code> or <code>colData(x)</code> that
contains the numeric vector</p></dd>


<dt id="arg-assay">assay<a class="anchor" aria-label="anchor" href="#arg-assay"></a></dt>
<dd><p>When a <strong>SummarizedExperiment</strong> is used, the index or name of
the assay</p></dd>


<dt id="arg-formula">formula<a class="anchor" aria-label="anchor" href="#arg-formula"></a></dt>
<dd><p>A formula specifying the classes on the left-hand side,
and the predictor terms on the right-hand side.</p></dd>


<dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>When a <strong>recipe</strong> or <strong>formula</strong> is used, <code>data</code> is specified as:</p><ul><li><p>A <strong>data frame</strong> containing both expression and classes, where columns
are the genes or class, and rows are the samples.</p></li>
</ul></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <code>clanc</code> object.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The original description of ClaNC can be found
<a href="10.1093/bioinformatics/bti681">here</a></p>
<p>While <code>active</code> sets the number of class-specific genes, each centroid will
have more than that number of genes. To explain by way of example, if <code>active = 5</code> and there are 3 classes, each centroid will have 15 genes, with 5 of
those genes being particular to a given class. If these genes are 'active' in
that class, their values will be the mean of the class. If the genes are not
active in that given class, their values will be the overall expression of
the given gene across all classes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">expression_matrix</span> <span class="op">&lt;-</span> <span class="va">synthetic_expression</span><span class="op">$</span><span class="va">expression</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">expression_matrix</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        sample1  sample2  sample3  sample4  sample5  sample6  sample7  sample8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene1 8.097529 7.119188 7.304400 7.554689 7.953206 7.714925 7.512700 8.597547</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2 8.641837 9.400416 8.500865 8.878687 8.318438 8.728683 7.812591 7.638167</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene3 3.436236 4.317915 3.435193 3.515755 3.024976 4.762209 5.048956 2.006646</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4 4.368008 5.212750 4.618249 4.201365 3.195294 4.707750 5.126769 6.178658</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene5 2.423974 3.563816 4.062362 2.163278 2.021435 2.813873 0.000000 4.652358</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene6 5.371205 5.919809 4.366915 4.805534 4.834856 5.622157 3.883531 3.593082</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        sample9 sample10 sample11 sample12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene1 6.475641 7.648858 8.637526 7.345038</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2 8.110285 7.906104 7.424728 7.927039</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene3 2.739211 3.111668 3.161077 4.306611</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4 5.170265 4.259578 5.872855 6.159023</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene5 1.532242 3.399823 3.691250 1.932937</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene6 4.246205 4.637316 3.575837 2.730452</span>
<span class="r-in"><span><span class="va">classes</span> <span class="op">&lt;-</span> <span class="va">synthetic_expression</span><span class="op">$</span><span class="va">classes</span></span></span>
<span class="r-in"><span><span class="va">classes</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] A A A A A A B B B B B B</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Levels: A B</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># data.frame/tibble/matrix interface:</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">clanc</span><span class="op">(</span><span class="va">expression_matrix</span>, classes <span class="op">=</span> <span class="va">classes</span>, active <span class="op">=</span> <span class="fl">5</span>, priors <span class="op">=</span> <span class="st">"equal"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;clanc&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $centroids</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    class   gene expression pooled_sd active prior</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1      A gene12   7.514718 0.4779155      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2      A  gene2   8.744821 0.3147537      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3      A gene13   8.936462 0.3418472      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4      A gene21   6.584681 0.5279636      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5      A gene24   4.307301 0.7214700      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6      A gene74   4.028507 0.4940783      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7      A gene41   4.328516 0.6317005      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8      A gene95   6.873184 0.4462475      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9      A gene52   3.743798 0.5173769      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10     A gene66   7.008174 0.5883218      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11     B gene12   8.072284 0.4779155      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12     B gene13   9.938137 0.3418472      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13     B  gene2   8.273987 0.3147537      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14     B gene24   3.370467 0.7214700      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15     B gene21   5.789423 0.5279636      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16     B gene41   5.518354 0.6317005      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17     B gene74   3.226598 0.4940783      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18     B gene52   2.438579 0.5173769      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19     B gene95   6.288173 0.4462475      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20     B gene66   7.891588 0.5883218      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Formula interface:</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Data must have class included as a column</span></span></span>
<span class="r-in"><span><span class="co"># Genes must be *columns* and samples must be *rows*</span></span></span>
<span class="r-in"><span><span class="co"># Hence the data transposition.</span></span></span>
<span class="r-in"><span><span class="va">for_formula</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>class <span class="op">=</span> <span class="va">classes</span>, <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">expression_matrix</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">clanc</span><span class="op">(</span><span class="va">class</span> <span class="op">~</span> <span class="va">.</span>, <span class="va">for_formula</span>, active <span class="op">=</span> <span class="fl">5</span>, priors <span class="op">=</span> <span class="st">"equal"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;clanc&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $centroids</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    class   gene expression pooled_sd active prior</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1      A gene12   7.514718 0.4779155      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2      A  gene2   8.744821 0.3147537      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3      A gene13   8.936462 0.3418472      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4      A gene21   6.584681 0.5279636      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5      A gene24   4.307301 0.7214700      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6      A gene74   4.028507 0.4940783      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7      A gene41   4.328516 0.6317005      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8      A gene95   6.873184 0.4462475      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9      A gene52   3.743798 0.5173769      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10     A gene66   7.008174 0.5883218      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11     B gene12   8.072284 0.4779155      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12     B gene13   9.938137 0.3418472      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13     B  gene2   8.273987 0.3147537      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14     B gene24   3.370467 0.7214700      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15     B gene21   5.789423 0.5279636      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16     B gene41   5.518354 0.6317005      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17     B gene74   3.226598 0.4940783      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18     B gene52   2.438579 0.5173769      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19     B gene95   6.288173 0.4462475      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20     B gene66   7.891588 0.5883218      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Recipes interface:</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">rec</span> <span class="op">&lt;-</span> <span class="fu">recipes</span><span class="fu">::</span><span class="fu"><a href="https://recipes.tidymodels.org/reference/recipe.html" class="external-link">recipe</a></span><span class="op">(</span><span class="va">class</span> <span class="op">~</span> <span class="va">.</span>, data <span class="op">=</span> <span class="va">for_formula</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">clanc</span><span class="op">(</span><span class="va">rec</span>, <span class="va">for_formula</span>, active <span class="op">=</span> <span class="fl">5</span>, priors <span class="op">=</span> <span class="st">"equal"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;clanc&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $centroids</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    class   gene expression pooled_sd active prior</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1      A gene12   7.514718 0.4779155      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2      A  gene2   8.744821 0.3147537      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3      A gene13   8.936462 0.3418472      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4      A gene21   6.584681 0.5279636      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5      A gene24   4.307301 0.7214700      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6      A gene74   4.028507 0.4940783      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7      A gene41   4.328516 0.6317005      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8      A gene95   6.873184 0.4462475      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9      A gene52   3.743798 0.5173769      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10     A gene66   7.008174 0.5883218      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11     B gene12   8.072284 0.4779155      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12     B gene13   9.938137 0.3418472      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13     B  gene2   8.273987 0.3147537      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14     B gene24   3.370467 0.7214700      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15     B gene21   5.789423 0.5279636      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16     B gene41   5.518354 0.6317005      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17     B gene74   3.226598 0.4940783      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18     B gene52   2.438579 0.5173769      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19     B gene95   6.288173 0.4462475      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20     B gene66   7.891588 0.5883218      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># SummarizedExperiment interface:</span></span></span>
<span class="r-in"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="va">expression_matrix</span>,</span></span>
<span class="r-in"><span>  colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    class <span class="op">=</span> <span class="va">classes</span>,</span></span>
<span class="r-in"><span>    active <span class="op">=</span> <span class="fl">5</span>,</span></span>
<span class="r-in"><span>    prior <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.5</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">clanc</span><span class="op">(</span><span class="va">se</span>, classes <span class="op">=</span> <span class="st">"class"</span>, active <span class="op">=</span> <span class="st">"active"</span>, priors <span class="op">=</span> <span class="st">"equal"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;clanc&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $centroids</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    class   gene expression pooled_sd active prior</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1      A gene12   7.514718 0.4779155      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2      A  gene2   8.744821 0.3147537      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3      A gene13   8.936462 0.3418472      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4      A gene21   6.584681 0.5279636      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5      A gene24   4.307301 0.7214700      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6      A gene74   4.028507 0.4940783      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7      A gene41   4.328516 0.6317005      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8      A gene95   6.873184 0.4462475      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9      A gene52   3.743798 0.5173769      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10     A gene66   7.008174 0.5883218      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11     B gene12   8.072284 0.4779155      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12     B gene13   9.938137 0.3418472      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13     B  gene2   8.273987 0.3147537      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14     B gene24   3.370467 0.7214700      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15     B gene21   5.789423 0.5279636      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16     B gene41   5.518354 0.6317005      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17     B gene74   3.226598 0.4940783      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18     B gene52   2.438579 0.5173769      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19     B gene95   6.288173 0.4462475      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20     B gene66   7.891588 0.5883218      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># ExpressionSet interface:</span></span></span>
<span class="r-in"><span><span class="va">adf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">expression_matrix</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  class <span class="op">=</span> <span class="va">classes</span></span></span>
<span class="r-in"><span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/class.AnnotatedDataFrame.html" class="external-link">AnnotatedDataFrame</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">es</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html" class="external-link">ExpressionSet</a></span><span class="op">(</span><span class="va">expression_matrix</span>, <span class="va">adf</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">clanc</span><span class="op">(</span><span class="va">es</span>, classes <span class="op">=</span> <span class="st">"class"</span>, active <span class="op">=</span> <span class="fl">5</span>, priors <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;clanc&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $centroids</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    class   gene expression pooled_sd active prior</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1      A gene12   7.514718 0.4779155      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2      A  gene2   8.744821 0.3147537      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3      A gene13   8.936462 0.3418472      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4      A gene21   6.584681 0.5279636      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5      A gene24   4.307301 0.7214700      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6      A gene74   4.028507 0.4940783      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7      A gene41   4.328516 0.6317005      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8      A gene95   6.873184 0.4462475      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9      A gene52   3.743798 0.5173769      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10     A gene66   7.008174 0.5883218      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11     B gene12   8.072284 0.4779155      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12     B gene13   9.938137 0.3418472      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13     B  gene2   8.273987 0.3147537      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14     B gene24   3.370467 0.7214700      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15     B gene21   5.789423 0.5279636      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16     B gene41   5.518354 0.6317005      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17     B gene74   3.226598 0.4940783      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18     B gene52   2.438579 0.5173769      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19     B gene95   6.288173 0.4462475      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20     B gene66   7.891588 0.5883218      5   0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Different numbers of active genes per class:</span></span></span>
<span class="r-in"><span><span class="co"># TODO</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Cross validation with `recipes`</span></span></span>
<span class="r-in"><span><span class="co"># TODO</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Using `tune` to find optimal `active`</span></span></span>
<span class="r-in"><span><span class="co"># TODO</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Kai Aragaki, Alan Dabney.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

