<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>case-study • reclanc</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="case-study">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">reclanc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/case-study.html">case-study</a></li>
    <li><a class="dropdown-item" href="../articles/using-reclanc.html">using-reclanc</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/KaiAragaki/reclanc/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>case-study</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/KaiAragaki/reclanc/blob/master/vignettes/case-study.Rmd" class="external-link"><code>vignettes/case-study.Rmd</code></a></small>
      <div class="d-none name"><code>case-study.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KaiAragaki/reclanc" class="external-link">reclanc</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cloudyr/aws.s3" class="external-link">aws.s3</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biobase" class="external-link">Biobase</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Let’s consider a relatively full-featured, practical use case for
<code>reclanc</code>. In this vignette, we’ll go over the basics of
fitting models, as well as how to leverage <code>tidymodels</code> to do
more elaborate things like resampling and tuning hyperparameters. We’ll
fit a final model, then use that to predict subtypes of an entirely new
dataset.</p>
<p>This vignette tries to assume very little knowledge about machine
learning or <code>tidymodels</code>.</p>
</div>
<div class="section level2">
<h2 id="fitting">Fitting<a class="anchor" aria-label="anchor" href="#fitting"></a>
</h2>
<div class="section level3">
<h3 id="a-simple-fit">A simple fit<a class="anchor" aria-label="anchor" href="#a-simple-fit"></a>
</h3>
<p>Let’s start with the fitting procedure. We first need gene expression
data.</p>
<p>The data I’m using is from <a href="https://pubmed.ncbi.nlm.nih.gov/22553347/" class="external-link">Sjödahl et
al. (2012)</a>. It contains RNA expression from 308 bladder cancer
tumors.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lund</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/aws.s3/man/s3saveRDS.html" class="external-link">s3readRDS</a></span><span class="op">(</span><span class="st">"lund.rds"</span>, <span class="st">"reclanc-lund"</span>, region <span class="op">=</span> <span class="st">"us-east-2"</span><span class="op">)</span></span>
<span><span class="va">lund</span></span>
<span><span class="co">#&gt; ExpressionSet (storageMode: lockedEnvironment)</span></span>
<span><span class="co">#&gt; assayData: 16940 features, 308 samples </span></span>
<span><span class="co">#&gt;   element names: exprs </span></span>
<span><span class="co">#&gt; protocolData: none</span></span>
<span><span class="co">#&gt; phenoData</span></span>
<span><span class="co">#&gt;   sampleNames: UC_0001_1 UC_0002_1 ... UC_0785_1 (308 total)</span></span>
<span><span class="co">#&gt;   varLabels: title source ... sample (16 total)</span></span>
<span><span class="co">#&gt;   varMetadata: labelDescription</span></span>
<span><span class="co">#&gt; featureData: none</span></span>
<span><span class="co">#&gt; experimentData: use 'experimentData(object)'</span></span>
<span><span class="co">#&gt; Annotation:</span></span></code></pre></div>
<p>In their paper, Sjödahl et al. used the transcriptional data to
classify the tumors into seven molecular subtypes (MS):</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">lund</span><span class="op">$</span><span class="va">molecular_subtype</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    MS1a    MS1b  MS2a.1  MS2a.2  MS2b.1 MS2b2.1 MS2b2.2 </span></span>
<span><span class="co">#&gt;      53      78      30      55      43      20      29</span></span></code></pre></div>
<p>We’d like to apply this subtype framework to other datasets. To do
this, we first need to generate centroids. Before we can begin, though,
we need to convert our outcomes to factors. In this case, our outcomes
are the molecular subtypes:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lund</span><span class="op">$</span><span class="va">molecular_subtype</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">lund</span><span class="op">$</span><span class="va">molecular_subtype</span><span class="op">)</span></span></code></pre></div>
<p>In its simplest form, since <code>clanc</code> accepts
<code>ExpressionSet</code> objects, we could do the following and be
done with it:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">simple_centroids</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clanc.html">clanc</a></span><span class="op">(</span><span class="va">lund</span>, classes <span class="op">=</span> <span class="st">"molecular_subtype"</span>, active <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">simple_centroids</span><span class="op">$</span><span class="va">centroids</span><span class="op">)</span></span>
<span><span class="co">#&gt;   class    gene expression pooled_sd active     prior</span></span>
<span><span class="co">#&gt; 1  MS1a   CXCL1   6.534490 0.8749133      5 0.1428571</span></span>
<span><span class="co">#&gt; 2  MS1a     MMD   7.922508 0.6429620      5 0.1428571</span></span>
<span><span class="co">#&gt; 3  MS1a C9orf19   8.378910 0.7510552      5 0.1428571</span></span>
<span><span class="co">#&gt; 4  MS1a    BNC1   5.297095 0.2106762      5 0.1428571</span></span>
<span><span class="co">#&gt; 5  MS1a  SLFN11   7.362887 0.6824663      5 0.1428571</span></span>
<span><span class="co">#&gt; 6  MS1a    CRAT   6.004517 0.3425669      5 0.1428571</span></span></code></pre></div>
<p>The problem with this method, though, is we have no idea if this is a
good fit or not. <code>active</code> is an argument that specifies the
number of genes that are used as distinguishing features for a given
class. In this case, each class will find 5 genes that have expression
patterns peculiar to that given molecular subtype, and each subtype will
have 7 (the total number of subtypes) x 5 (number of active genes) = 35
genes in it (see my <a href="https://kai.rbind.io/posts/projects-reclanc/" class="external-link">blog post</a> or -
better yet - <a href="http://www.ncbi.nlm.nih.gov/pubmed/16174683" class="external-link">the
original paper</a> for more details). Could we have gotten a better fit
with more genes? Are we selecting more genes than we need? How would we
know?</p>
</div>
<div class="section level3">
<h3 id="setting-the-stage-for-more-elaborate-analyses">Setting the stage for more elaborate analyses<a class="anchor" aria-label="anchor" href="#setting-the-stage-for-more-elaborate-analyses"></a>
</h3>
<p>Before we can get started on tackling these larger questions, let’s
take a brief detour to the land of <a href="https://www.tidymodels.org/" class="external-link"><code>tidymodels</code></a>.
<code>tidymodels</code> is a collection of packages that make running
and tuning algorithms like this much less painful and much more
standardized.</p>
<p>In order to leverage <code>tidymodels</code>, we need to buy-in to
their data structures.</p>
<p>(Aside: I don’t mean to make the buy-in sound begrudging. When I say
need, I really mean it: we’re going to be specifying very long formulas,
which for some reason R really, <em>really</em> hates. Emil Hvitfeldt
recently (at time of writing) has <a href="https://github.com/tidymodels/recipes/pull/1283" class="external-link">allowed
<code>tidymodels</code> to handle long formulas gracefully</a>, so using
<code>tidymodels</code> infrastructure is a gift, not a chore.)</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidymodels.tidymodels.org" class="external-link">tidymodels</a></span><span class="op">)</span></span></code></pre></div>
<p>Many <code>tidymodels</code> workflows begin with a <em>model
specification</em>. The rationale behind this is to separate the
<em>model specification</em> step from the <em>model fitting</em> step
(whereas in base R, they generally all happen at once).
<code>reclanc</code> makes it easy to specify a model by adding a custom
engine to <code><a href="https://parsnip.tidymodels.org/reference/discrim_linear.html" class="external-link">parsnip::discrim_linear</a></code>, so specifying a model
looks like this:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu">discrim_linear</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">set_engine</span><span class="op">(</span></span>
<span>    engine <span class="op">=</span> <span class="st">"clanc"</span>, <span class="co"># Note: "clanc", not "reclanc"</span></span>
<span>    active <span class="op">=</span> <span class="fl">5</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>This <code>mod</code> doesn’t do anything - and that’s kind of the
point: it only specifies the model we will later fit with, but doesn’t
do any fitting itself. This allows us to reuse the specification across
our code.</p>
<p>The next step is to wrangle our data a bit to be in a ‘wide’ format,
where all columns are outcomes (classes) and predictors (genes), and all
rows are observations (samples):</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wrangled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>class <span class="op">=</span> <span class="va">lund</span><span class="op">$</span><span class="va">molecular_subtype</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">lund</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">wrangled</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;            class LOC23117   FCGR2B    TRIM44 C15orf39</span></span>
<span><span class="co">#&gt; UC_0001_1   MS1b 5.565262 5.306654  9.305053 6.430063</span></span>
<span><span class="co">#&gt; UC_0002_1 MS2b.1 5.505854 5.731128  9.242790 7.265748</span></span>
<span><span class="co">#&gt; UC_0003_1 MS2a.2 5.336140 5.540470  9.888668 7.244976</span></span>
<span><span class="co">#&gt; UC_0006_2 MS2b.1 5.576748 5.847743  9.408895 7.377358</span></span>
<span><span class="co">#&gt; UC_0007_1 MS2a.2 5.414919 5.510507 10.482469 6.435552</span></span>
<span><span class="co">#&gt; UC_0008_1 MS2b.1 5.279174 5.633093  9.112754 7.057977</span></span></code></pre></div>
<p>Finally, we specify a formula for fitting the model. This uses the
<code>recipes</code> package from <code>tidymodels</code>. While this is
a delightful package that can help you preprocess your data, it’s out of
the scope of this vignette. Instead, just think of it as a way to
specify a formula that keeps R from blowing up:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Note that the recipe requires 'template data'</span></span>
<span><span class="va">recipe</span> <span class="op">&lt;-</span> <span class="fu">recipe</span><span class="op">(</span><span class="va">class</span> <span class="op">~</span> <span class="va">.</span>, <span class="va">wrangled</span><span class="op">)</span> </span></code></pre></div>
<p>We can bundle our model specification (<code>mod</code>) and our
preprocessing steps (<code>recipe</code>, which is just a formula) into
a <code>workflow</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wf</span> <span class="op">&lt;-</span> <span class="fu">workflow</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">add_recipe</span><span class="op">(</span><span class="va">recipe</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">add_model</span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span>
<span><span class="va">wf</span></span>
<span><span class="co">#&gt; ══ Workflow ════════════════════════════════════════════════════════════════════</span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">Preprocessor:</span> Recipe</span></span>
<span><span class="co">#&gt; <span style="font-style: italic;">Model:</span> discrim_linear()</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Preprocessor ────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; 0 Recipe Steps</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Model ───────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; Linear Discriminant Model Specification (classification)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Engine-Specific Arguments:</span></span>
<span><span class="co">#&gt;   active = 5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Computational engine: clanc</span></span></code></pre></div>
<p>Now we can fit our model:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tidymodels_fit</span> <span class="op">&lt;-</span> <span class="fu">fit</span><span class="op">(</span><span class="va">wf</span>, data <span class="op">=</span> <span class="va">wrangled</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tidymodels_fit</span><span class="op">$</span><span class="va">fit</span><span class="op">$</span><span class="va">centroids</span><span class="op">)</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
<p>You’ll notice that our results are the same as what we saw
previously, demonstrating that while we’re using <code>tidymodels</code>
rather than base R, we’re still doing the same thing.</p>
</div>
<div class="section level3">
<h3 id="measuring-fit-accuracy-with-cross-validation">Measuring fit accuracy with cross-validation<a class="anchor" aria-label="anchor" href="#measuring-fit-accuracy-with-cross-validation"></a>
</h3>
<p>Now that we’ve dialed in to the <code>tidymodels</code> framework, we
can do a lot of elaborate things with ease. One of our concerns is
whether 5 active genes was a good choice (<code>active = 5</code>). A
somewhat simple way to determine how good our choice of 5 genes is to
use <a href="https://en.wikipedia.org/wiki/Cross-validation_(statistics)" class="external-link">cross-validation</a>.
Cross-validation allows us to test how good our fit is by training our
model on, say, 80% of our data, and testing it on the rest (see the
Wikipedia diagram of a k-fold cross validation). This allows us to get a
measure of how good our fit is, without having to break out our actual
test data - which in general should only be used when we’re ready to
finalize our model.</p>
<p>Speaking of test data, let’s go ahead and split that off now. We’ll
lock our test data away and only use it once we’ve fit our final model.
Until then, we’ll use cross validation to assess how good the fit is,
essentially using our training data as its own testing data.</p>
<p>Of course, <code>tidymodels</code> makes this easy too, by using
<code><a href="https://rsample.tidymodels.org/reference/initial_split.html" class="external-link">rsample::initial_split</a></code>:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">splits</span> <span class="op">&lt;-</span> <span class="fu">initial_split</span><span class="op">(</span><span class="va">wrangled</span>, prop <span class="op">=</span> <span class="fl">0.8</span>, strata <span class="op">=</span> <span class="va">class</span><span class="op">)</span></span>
<span><span class="va">train</span> <span class="op">&lt;-</span> <span class="fu">training</span><span class="op">(</span><span class="va">splits</span><span class="op">)</span></span>
<span><span class="va">test</span> <span class="op">&lt;-</span> <span class="fu">testing</span><span class="op">(</span><span class="va">splits</span><span class="op">)</span></span></code></pre></div>
<p><code>train</code> and <code>test</code> are just subsets of the
original data, containing 80% and 20% of the original data
(respectively). It also tries to maintain the relative proportions of
each of the classes within each of the datasets (because we set
<code>strata = class</code>):</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/proportions.html" class="external-link">prop.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">train</span><span class="op">$</span><span class="va">class</span><span class="op">)</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    MS1a    MS1b  MS2a.1  MS2a.2  MS2b.1 MS2b2.1 MS2b2.2 </span></span>
<span><span class="co">#&gt;    0.17    0.25    0.10    0.18    0.15    0.07    0.08</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/proportions.html" class="external-link">prop.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">test</span><span class="op">$</span><span class="va">class</span><span class="op">)</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    MS1a    MS1b  MS2a.1  MS2a.2  MS2b.1 MS2b2.1 MS2b2.2 </span></span>
<span><span class="co">#&gt;    0.19    0.27    0.08    0.16    0.11    0.05    0.16</span></span></code></pre></div>
<p>Creating folds for cross validation is nearly the same as
<code>initial_split</code>:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">folds</span> <span class="op">&lt;-</span> <span class="fu">vfold_cv</span><span class="op">(</span><span class="va">train</span>, v <span class="op">=</span> <span class="fl">5</span>, strata <span class="op">=</span> <span class="va">class</span><span class="op">)</span></span>
<span><span class="va">folds</span></span>
<span><span class="co">#&gt; #  5-fold cross-validation using stratification </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 2</span></span></span>
<span><span class="co">#&gt;   splits           id   </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> <span style="color: #949494;">&lt;split [193/51]&gt;</span> Fold1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> <span style="color: #949494;">&lt;split [193/51]&gt;</span> Fold2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> <span style="color: #949494;">&lt;split [195/49]&gt;</span> Fold3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> <span style="color: #949494;">&lt;split [197/47]&gt;</span> Fold4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> <span style="color: #949494;">&lt;split [198/46]&gt;</span> Fold5</span></span></code></pre></div>
<p>We can reuse our workflow <code>wf</code>, which contains our model
and formula. The only difference is that we use
<code>fit_resamples</code>, and we specify a metric we want to use to
measure how good our fit is (remember that every fold has a chunk of
data it uses to test the fit). For simplicity, let’s use accuracy:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fits</span> <span class="op">&lt;-</span> <span class="fu">fit_resamples</span><span class="op">(</span></span>
<span>  <span class="va">wf</span>,</span>
<span>  <span class="va">folds</span>,</span>
<span>  metrics <span class="op">=</span> <span class="fu">metric_set</span><span class="op">(</span><span class="va">accuracy</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 35/35 (100%) genes in centroids found in `new_data`</span></span>
<span><span class="co">#&gt; 35/35 (100%) genes in centroids found in `new_data`</span></span>
<span><span class="co">#&gt; 35/35 (100%) genes in centroids found in `new_data`</span></span>
<span><span class="co">#&gt; 35/35 (100%) genes in centroids found in `new_data`</span></span>
<span><span class="co">#&gt; 35/35 (100%) genes in centroids found in `new_data`</span></span>
<span><span class="va">fits</span></span>
<span><span class="co">#&gt; # Resampling results</span></span>
<span><span class="co">#&gt; # 5-fold cross-validation using stratification </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 4</span></span></span>
<span><span class="co">#&gt;   splits           id    .metrics         .notes          </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> <span style="color: #949494;">&lt;split [193/51]&gt;</span> Fold1 <span style="color: #949494;">&lt;tibble [1 × 4]&gt;</span> <span style="color: #949494;">&lt;tibble [0 × 3]&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> <span style="color: #949494;">&lt;split [193/51]&gt;</span> Fold2 <span style="color: #949494;">&lt;tibble [1 × 4]&gt;</span> <span style="color: #949494;">&lt;tibble [0 × 3]&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> <span style="color: #949494;">&lt;split [195/49]&gt;</span> Fold3 <span style="color: #949494;">&lt;tibble [1 × 4]&gt;</span> <span style="color: #949494;">&lt;tibble [0 × 3]&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> <span style="color: #949494;">&lt;split [197/47]&gt;</span> Fold4 <span style="color: #949494;">&lt;tibble [1 × 4]&gt;</span> <span style="color: #949494;">&lt;tibble [0 × 3]&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> <span style="color: #949494;">&lt;split [198/46]&gt;</span> Fold5 <span style="color: #949494;">&lt;tibble [1 × 4]&gt;</span> <span style="color: #949494;">&lt;tibble [0 × 3]&gt;</span></span></span></code></pre></div>
<p>We can then extract our accuracy metrics by using
<code>collect_metrics</code>, which roots around in each of our fits and
helpfully extracts the metrics, aggregates them, and calculated the
standard error:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metrics</span> <span class="op">&lt;-</span> <span class="fu">collect_metrics</span><span class="op">(</span><span class="va">fits</span><span class="op">)</span></span>
<span><span class="va">metrics</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 6</span></span></span>
<span><span class="co">#&gt;   .metric  .estimator  mean     n std_err .config             </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> accuracy multiclass 0.737     5  0.028<span style="text-decoration: underline;">9</span> Preprocessor1_Model1</span></span></code></pre></div>
<p>Our model has an accuracy of about 74%. Applying this model to our
testing data:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fit a model using *all* of our training data</span></span>
<span><span class="va">final_fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clanc.html">clanc</a></span><span class="op">(</span><span class="va">class</span> <span class="op">~</span> <span class="va">.</span>, <span class="va">train</span>, active <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use it to predict the (known) classes of our test data</span></span>
<span><span class="va">preds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">final_fit</span>, new_data <span class="op">=</span> <span class="va">test</span>, type <span class="op">=</span> <span class="st">"class"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 35/35 (100%) genes in centroids found in `new_data`</span></span>
<span><span class="va">w_preds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">preds</span>, <span class="va">test</span><span class="op">)</span></span>
<span><span class="co"># Compare known class vs predicted class</span></span>
<span><span class="va">metric</span> <span class="op">&lt;-</span> <span class="fu">accuracy</span><span class="op">(</span><span class="va">w_preds</span>, <span class="va">class</span>, <span class="va">.pred_class</span><span class="op">)</span></span>
<span><span class="va">metric</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 3</span></span></span>
<span><span class="co">#&gt;   .metric  .estimator .estimate</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> accuracy multiclass     0.734</span></span></code></pre></div>
<p>Note that our testing data accuracy (%) approximates the training
data accuracy (74%).</p>
</div>
<div class="section level3">
<h3 id="tuning-hyperparameters-with-tune">Tuning hyperparameters with <code>tune</code><a class="anchor" aria-label="anchor" href="#tuning-hyperparameters-with-tune"></a>
</h3>
<p>Now we at least have <em>some</em> measure of how good our model
fits, but could it be better with more genes? Could we get away with
fewer? Running the same command over and over again with different
numbers is a drag - fortunately, there’s yet another beautiful package
to help us: <code>tune</code>.</p>
<p>To use <code>tune</code>, we need to re-specify our model to let
<code>tune</code> know what parameters we want to tune:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tune_mod</span> <span class="op">&lt;-</span> <span class="fu">discrim_linear</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">set_engine</span><span class="op">(</span></span>
<span>    engine <span class="op">=</span> <span class="st">"clanc"</span>,</span>
<span>    active <span class="op">=</span> <span class="fu">tune</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>We could update our previous workflow using
<code>update_model</code>, but let’s just declare a new one:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tune_wf</span> <span class="op">&lt;-</span> <span class="fu">workflow</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">add_recipe</span><span class="op">(</span><span class="va">recipe</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">add_model</span><span class="op">(</span><span class="va">tune_mod</span><span class="op">)</span></span></code></pre></div>
<p>We then have to specify a range of values of <code>active</code> to
try:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">values</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>active <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>from <span class="op">=</span> <span class="fl">1</span>, to <span class="op">=</span> <span class="fl">50</span>, by <span class="op">=</span> <span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">values</span></span>
<span><span class="co">#&gt;    active</span></span>
<span><span class="co">#&gt; 1       1</span></span>
<span><span class="co">#&gt; 2       5</span></span>
<span><span class="co">#&gt; 3       9</span></span>
<span><span class="co">#&gt; 4      13</span></span>
<span><span class="co">#&gt; 5      17</span></span>
<span><span class="co">#&gt; 6      21</span></span>
<span><span class="co">#&gt; 7      25</span></span>
<span><span class="co">#&gt; 8      29</span></span>
<span><span class="co">#&gt; 9      33</span></span>
<span><span class="co">#&gt; 10     37</span></span>
<span><span class="co">#&gt; 11     41</span></span>
<span><span class="co">#&gt; 12     45</span></span>
<span><span class="co">#&gt; 13     49</span></span></code></pre></div>
<p>We can then fit our <code>folds</code> using the spread of values we
chose:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This is going to take some time, since we're fitting 5 folds 13 times each.</span></span>
<span><span class="va">tuned</span> <span class="op">&lt;-</span> <span class="fu">tune_grid</span><span class="op">(</span></span>
<span>  <span class="va">tune_wf</span>,</span>
<span>  <span class="va">folds</span>,</span>
<span>  metrics <span class="op">=</span> <span class="fu">metric_set</span><span class="op">(</span><span class="va">accuracy</span><span class="op">)</span>,</span>
<span>  grid <span class="op">=</span> <span class="va">values</span></span>
<span><span class="op">)</span></span>
<span><span class="va">tuned</span></span>
<span><span class="co">#&gt; # Tuning results</span></span>
<span><span class="co">#&gt; # 5-fold cross-validation using stratification </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 4</span></span></span>
<span><span class="co">#&gt;   splits           id    .metrics          .notes          </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> <span style="color: #949494;">&lt;split [193/51]&gt;</span> Fold1 <span style="color: #949494;">&lt;tibble [13 × 5]&gt;</span> <span style="color: #949494;">&lt;tibble [0 × 3]&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> <span style="color: #949494;">&lt;split [193/51]&gt;</span> Fold2 <span style="color: #949494;">&lt;tibble [13 × 5]&gt;</span> <span style="color: #949494;">&lt;tibble [0 × 3]&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> <span style="color: #949494;">&lt;split [195/49]&gt;</span> Fold3 <span style="color: #949494;">&lt;tibble [13 × 5]&gt;</span> <span style="color: #949494;">&lt;tibble [0 × 3]&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> <span style="color: #949494;">&lt;split [197/47]&gt;</span> Fold4 <span style="color: #949494;">&lt;tibble [13 × 5]&gt;</span> <span style="color: #949494;">&lt;tibble [0 × 3]&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> <span style="color: #949494;">&lt;split [198/46]&gt;</span> Fold5 <span style="color: #949494;">&lt;tibble [13 × 5]&gt;</span> <span style="color: #949494;">&lt;tibble [0 × 3]&gt;</span></span></span></code></pre></div>
<p>As before, we can collect our metrics - this time, however, we have a
summary of metrics for each of values for <code>active</code>:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tuned_metrics</span> <span class="op">&lt;-</span> <span class="fu">collect_metrics</span><span class="op">(</span><span class="va">tuned</span><span class="op">)</span></span>
<span><span class="va">tuned_metrics</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 13 × 7</span></span></span>
<span><span class="co">#&gt;    active .metric  .estimator  mean     n std_err .config              </span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>      1 accuracy multiclass 0.585     5  0.036<span style="text-decoration: underline;">8</span> Preprocessor1_Model01</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>      5 accuracy multiclass 0.737     5  0.028<span style="text-decoration: underline;">9</span> Preprocessor1_Model02</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>      9 accuracy multiclass 0.748     5  0.049<span style="text-decoration: underline;">6</span> Preprocessor1_Model03</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>     13 accuracy multiclass 0.781     5  0.040<span style="text-decoration: underline;">3</span> Preprocessor1_Model04</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>     17 accuracy multiclass 0.770     5  0.028<span style="text-decoration: underline;">0</span> Preprocessor1_Model05</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     21 accuracy multiclass 0.774     5  0.033<span style="text-decoration: underline;">5</span> Preprocessor1_Model06</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>     25 accuracy multiclass 0.785     5  0.037<span style="text-decoration: underline;">8</span> Preprocessor1_Model07</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>     29 accuracy multiclass 0.794     5  0.031<span style="text-decoration: underline;">9</span> Preprocessor1_Model08</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>     33 accuracy multiclass 0.773     5  0.028<span style="text-decoration: underline;">1</span> Preprocessor1_Model09</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>     37 accuracy multiclass 0.790     5  0.029<span style="text-decoration: underline;">5</span> Preprocessor1_Model10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     41 accuracy multiclass 0.794     5  0.033<span style="text-decoration: underline;">9</span> Preprocessor1_Model11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span>     45 accuracy multiclass 0.815     5  0.026<span style="text-decoration: underline;">7</span> Preprocessor1_Model12</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span>     49 accuracy multiclass 0.815     5  0.027<span style="text-decoration: underline;">7</span> Preprocessor1_Model13</span></span></code></pre></div>
<p>Or graphically:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">tuned_metrics</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">active</span>, <span class="va">mean</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">coord_cartesian</span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Number Active Genes"</span>, y <span class="op">=</span> <span class="st">"Accuracy"</span><span class="op">)</span></span></code></pre></div>
<p><img src="case-study_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
<p>It looks like we read maximal accuracy at around 21 genes - let’s
choose 20 genes for a nice round number:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">final_fit_tuned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clanc.html">clanc</a></span><span class="op">(</span><span class="va">class</span> <span class="op">~</span> <span class="va">.</span>, data <span class="op">=</span> <span class="va">train</span>, active <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="co"># Use it to predict the (known) classes of our test data:</span></span>
<span><span class="va">preds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">final_fit_tuned</span>, new_data <span class="op">=</span> <span class="va">test</span>, type <span class="op">=</span> <span class="st">"class"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 140/140 (100%) genes in centroids found in `new_data`</span></span>
<span><span class="va">w_preds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">preds</span>, <span class="va">test</span><span class="op">)</span></span>
<span><span class="co"># Compare known class vs predicted class:</span></span>
<span><span class="va">metric</span> <span class="op">&lt;-</span> <span class="fu">accuracy</span><span class="op">(</span><span class="va">w_preds</span>, <span class="va">class</span>, <span class="va">.pred_class</span><span class="op">)</span></span>
<span><span class="va">metric</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 3</span></span></span>
<span><span class="co">#&gt;   .metric  .estimator .estimate</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> accuracy multiclass     0.812</span></span></code></pre></div>
<p>It looks like our accuracy is a little better now that we’ve chosen
an optimal number of active genes.</p>
</div>
</div>
<div class="section level2">
<h2 id="predicting">Predicting<a class="anchor" aria-label="anchor" href="#predicting"></a>
</h2>
<p>Now we want to apply our classifier to new data. Our second dataset
is <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE97768" class="external-link">RNAseq
data from 30 bladder cancer cell lines</a>:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">cellebrate</span><span class="op">)</span></span>
<span><span class="va">cell_rna</span></span>
<span><span class="co">#&gt; class: DESeqDataSet </span></span>
<span><span class="co">#&gt; dim: 18548 30 </span></span>
<span><span class="co">#&gt; metadata(1): version</span></span>
<span><span class="co">#&gt; assays(2): counts rlog_norm_counts</span></span>
<span><span class="co">#&gt; rownames(18548): TSPAN6 TNMD ... MT-ND5 MT-ND6</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(30): 1A6 253JP ... UC7 UC9</span></span>
<span><span class="co">#&gt; colData names(5): cell bsl lum call clade</span></span></code></pre></div>
<p>Predicting is incredibly simple. Since we’re using a different
sequencing method (RNAseq vs array-based sequencing), it probably makes
sense to use a correlation based classification rather than the original
distance-based metric used in the original ClaNC package. We can do that
by specifying <code>type = "numeric"</code> and then whatever
correlation method we prefer.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell_preds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span></span>
<span>  <span class="va">final_fit_tuned</span>,</span>
<span>  <span class="va">cell_rna</span>,</span>
<span>  assay <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"numeric"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spearman"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 118/140 (84%) genes in centroids found in `new_data`</span></span>
<span></span>
<span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cell_rna</span><span class="op">)</span>, <span class="va">cell_preds</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">out</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 30 × 12</span></span></span>
<span><span class="co">#&gt;    cell     bsl    lum call  clade            .pred_MS1a .pred_MS1b .pred_MS2a.1</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 1A6     99.0   1.02 BSL   Epithelial Other     0.060<span style="text-decoration: underline;">0</span>      0.224        0.149</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 253JP   76.6  23.4  BSL   Unknown              0.057<span style="text-decoration: underline;">4</span>      0.240        0.219</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 5637    98.5   1.46 BSL   Epithelial Other     0.095<span style="text-decoration: underline;">8</span>      0.243        0.160</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> BV      49.9  50.1  LUM   Unknown              0.075<span style="text-decoration: underline;">8</span>      0.262        0.238</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> HT1197  56.0  44.0  BSL   Epithelial Other     0.119       0.288        0.224</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> HT1376  10.9  89.1  LUM   Epithelial Other     0.100       0.277        0.238</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> J82     98.1   1.91 BSL   Mesenchymal          0.127       0.292        0.219</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> RT112    0   100    LUM   Luminal Papilla…     0.173       0.380        0.294</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> RT4      0   100    LUM   Luminal Papilla…     0.134       0.317        0.257</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> RT4V6    0   100    LUM   Luminal Papilla…     0.143       0.207        0.165</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 20 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4 more variables: .pred_MS2a.2 &lt;dbl&gt;, .pred_MS2b.1 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   .pred_MS2b2.1 &lt;dbl&gt;, .pred_MS2b2.2 &lt;dbl&gt;</span></span></span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plotting_data</span> <span class="op">&lt;-</span> <span class="va">out</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">starts_with</span><span class="op">(</span><span class="st">".pred"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plotting_data</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">cell</span>, <span class="va">value</span>, color <span class="op">=</span> <span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="op">~</span><span class="va">clade</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, space <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span></span></code></pre></div>
<p><img src="case-study_files/figure-html/unnamed-chunk-28-1.png" width="960"></p>
<p>In the Sjödahl paper, the seven subtypes were simplified into five
subtypes by merging some of the two that had similar biological pathways
activated. To ease interpretation, we can do that too:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">table</span> <span class="op">&lt;-</span> <span class="va">plotting_data</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>winner <span class="op">=</span> <span class="va">name</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="va">value</span><span class="op">)</span><span class="op">]</span>, .by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cell</span>, <span class="va">clade</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    five <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span></span>
<span>      <span class="va">winner</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">".pred_MS1a"</span>, <span class="st">".pred_MS1b"</span><span class="op">)</span> <span class="op">~</span> <span class="st">"Urobasal A"</span>,</span>
<span>      <span class="va">winner</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">".pred_MS2a.1"</span>, <span class="st">".pred_MS2a.2"</span><span class="op">)</span> <span class="op">~</span> <span class="st">"Genomically unstable"</span>,</span>
<span>      <span class="va">winner</span> <span class="op">==</span> <span class="st">".pred_MS2b.1"</span> <span class="op">~</span> <span class="st">"Infiltrated"</span>,</span>
<span>      <span class="va">winner</span> <span class="op">==</span> <span class="st">".pred_MS2b2.1"</span> <span class="op">~</span> <span class="st">"Uro-B"</span>,</span>
<span>      <span class="va">winner</span> <span class="op">==</span> <span class="st">".pred_MS2b2.2"</span> <span class="op">~</span> <span class="st">"SCC-like"</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">relocate</span><span class="op">(</span><span class="va">cell</span>, <span class="va">five</span>, <span class="va">clade</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">table</span>, n <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 30 × 4</span></span></span>
<span><span class="co">#&gt;    cell   five                 clade             winner       </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 1A6    SCC-like             Epithelial Other  .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 253JP  SCC-like             Unknown           .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 5637   SCC-like             Epithelial Other  .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> BV     Urobasal A           Unknown           .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> HT1197 SCC-like             Epithelial Other  .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> HT1376 SCC-like             Epithelial Other  .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> J82    Urobasal A           Mesenchymal       .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> RT112  Urobasal A           Luminal Papillary .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> RT4    Urobasal A           Luminal Papillary .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> RT4V6  Urobasal A           Luminal Papillary .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> SCaBER SCC-like             Epithelial Other  .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> SW780  Urobasal A           Luminal Papillary .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> T24    SCC-like             Mesenchymal       .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> TCCSup SCC-like             Mesenchymal       .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> UC10   SCC-like             Epithelial Other  .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> UC11   SCC-like             Mesenchymal       .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> UC12   Urobasal A           Mesenchymal       .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> UC13   SCC-like             Mesenchymal       .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> UC14   Urobasal A           Luminal Papillary .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">20</span> UC15   SCC-like             Epithelial Other  .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">21</span> UC16   SCC-like             Epithelial Other  .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">22</span> UC17   SCC-like             Luminal Papillary .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">23</span> UC18   SCC-like             Mesenchymal       .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">24</span> UC1    Urobasal A           Luminal Papillary .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">25</span> UC3    SCC-like             Mesenchymal       .pred_MS2b2.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">26</span> UC4    Urobasal A           Unknown           .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">27</span> UC5    Urobasal A           Luminal Papillary .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">28</span> UC6    Urobasal A           Luminal Papillary .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">29</span> UC7    Urobasal A           Epithelial Other  .pred_MS1b   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">30</span> UC9    Genomically unstable Epithelial Other  .pred_MS2a.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Kai Aragaki, Alan Dabney.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
