<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>using-reclanc • reclanc</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="using-reclanc">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">reclanc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/case-study.html">case-study</a></li>
    <li><a class="dropdown-item" href="../articles/using-reclanc.html">using-reclanc</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/KaiAragaki/reclanc/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>using-reclanc</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/KaiAragaki/reclanc/blob/master/vignettes/using-reclanc.Rmd" class="external-link"><code>vignettes/using-reclanc.Rmd</code></a></small>
      <div class="d-none name"><code>using-reclanc.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette will provide a brief introduction into the basic
<em>usage</em> of <code>reclanc</code>. If you’re interested in how
<code>reclanc</code> works, I’d recommend reading the <a href="https://kai.rbind.io/posts/projects-reclanc/" class="external-link">blog post</a> I
wrote and the <a href="https://academic.oup.com/bioinformatics/article/21/22/4148/194954?login=false" class="external-link">original
paper</a> by Alan Dabney, who created the original <code>ClaNC</code>
algorithm.</p>
</div>
<div class="section level2">
<h2 id="fitting">Fitting<a class="anchor" aria-label="anchor" href="#fitting"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KaiAragaki/reclanc" class="external-link">reclanc</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/tidymodels/parsnip" class="external-link">parsnip</a></span><span class="op">)</span></span></code></pre></div>
<p>To create new centroids from existing expression data, use the
<code>clanc</code> function.</p>
<p><code>reclanc</code> provides some synthetic expression data that we
can work with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">synthetic_expression</span>, <span class="va">head</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; $expression</span></span>
<span><span class="co">#&gt;        sample1  sample2  sample3  sample4  sample5  sample6  sample7  sample8</span></span>
<span><span class="co">#&gt; gene1 8.097529 7.119188 7.304400 7.554689 7.953206 7.714925 7.512700 8.597547</span></span>
<span><span class="co">#&gt; gene2 8.641837 9.400416 8.500865 8.878687 8.318438 8.728683 7.812591 7.638167</span></span>
<span><span class="co">#&gt; gene3 3.436236 4.317915 3.435193 3.515755 3.024976 4.762209 5.048956 2.006646</span></span>
<span><span class="co">#&gt; gene4 4.368008 5.212750 4.618249 4.201365 3.195294 4.707750 5.126769 6.178658</span></span>
<span><span class="co">#&gt; gene5 2.423974 3.563816 4.062362 2.163278 2.021435 2.813873 0.000000 4.652358</span></span>
<span><span class="co">#&gt; gene6 5.371205 5.919809 4.366915 4.805534 4.834856 5.622157 3.883531 3.593082</span></span>
<span><span class="co">#&gt;        sample9 sample10 sample11 sample12</span></span>
<span><span class="co">#&gt; gene1 6.475641 7.648858 8.637526 7.345038</span></span>
<span><span class="co">#&gt; gene2 8.110285 7.906104 7.424728 7.927039</span></span>
<span><span class="co">#&gt; gene3 2.739211 3.111668 3.161077 4.306611</span></span>
<span><span class="co">#&gt; gene4 5.170265 4.259578 5.872855 6.159023</span></span>
<span><span class="co">#&gt; gene5 1.532242 3.399823 3.691250 1.932937</span></span>
<span><span class="co">#&gt; gene6 4.246205 4.637316 3.575837 2.730452</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $classes</span></span>
<span><span class="co">#&gt; [1] A A A A A A</span></span>
<span><span class="co">#&gt; Levels: A B</span></span></code></pre>
<p>These data include 12 samples, 6 of class “A” and 6 of class “B”,
with 100 genes.</p>
<p><code>reclanc</code> is able to ingest a variety of common formats
for expression data. Objects like <code>SummarizedExperiments</code> and
<code>ExpressionSets</code> are frequently used in bioinformatic
analyses and arrange their data so samples are in columns and genes are
in rows. This is in conflict with the expected formula input in base R,
where predictors (genes) and outcomes (classes) are in columns.
<code>reclanc</code> eases this friction by expecting input in its most
common format, abstracting away the wrangling aspect of analysis.</p>
<p>Because of this, there are two broad categories of input - ‘wide
data’ and ‘tall data’.</p>
<div class="section level3">
<h3 id="wide-inputs">Wide inputs<a class="anchor" aria-label="anchor" href="#wide-inputs"></a>
</h3>
<p>Wide inputs require data that has both predictors and outcomes as
columns, together, in a single <code>data.frame</code>.</p>
<div class="section level4">
<h4 id="formula">Formula<a class="anchor" aria-label="anchor" href="#formula"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">form_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span></span>
<span>  class <span class="op">=</span> <span class="va">synthetic_expression</span><span class="op">$</span><span class="va">classes</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">synthetic_expression</span><span class="op">$</span><span class="va">expression</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">form_data</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;         class    gene1    gene2    gene3    gene4</span></span>
<span><span class="co">#&gt; sample1     A 8.097529 8.641837 3.436236 4.368008</span></span>
<span><span class="co">#&gt; sample2     A 7.119188 9.400416 4.317915 5.212750</span></span>
<span><span class="co">#&gt; sample3     A 7.304400 8.500865 3.435193 4.618249</span></span>
<span><span class="co">#&gt; sample4     A 7.554689 8.878687 3.515755 4.201365</span></span>
<span><span class="co">#&gt; sample5     A 7.953206 8.318438 3.024976 3.195294</span></span>
<span><span class="co">#&gt; sample6     A 7.714925 8.728683 4.762209 4.707750</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/clanc.html">clanc</a></span><span class="op">(</span><span class="va">class</span> <span class="op">~</span> <span class="va">.</span>, <span class="va">form_data</span>, active <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; &lt;clanc&gt; </span></span>
<span><span class="co">#&gt; $centroids</span></span>
<span><span class="co">#&gt;    class   gene expression pooled_sd active prior</span></span>
<span><span class="co">#&gt; 1      A gene12   7.514718 0.4779155      5   0.5</span></span>
<span><span class="co">#&gt; 2      A  gene2   8.744821 0.3147537      5   0.5</span></span>
<span><span class="co">#&gt; 3      A gene13   8.936462 0.3418472      5   0.5</span></span>
<span><span class="co">#&gt; 4      A gene21   6.584681 0.5279636      5   0.5</span></span>
<span><span class="co">#&gt; 5      A gene24   4.307301 0.7214700      5   0.5</span></span>
<span><span class="co">#&gt; 6      A gene74   4.028507 0.4940783      5   0.5</span></span>
<span><span class="co">#&gt; 7      A gene41   4.328516 0.6317005      5   0.5</span></span>
<span><span class="co">#&gt; 8      A gene95   6.873184 0.4462475      5   0.5</span></span>
<span><span class="co">#&gt; 9      A gene52   3.743798 0.5173769      5   0.5</span></span>
<span><span class="co">#&gt; 10     A gene66   7.008174 0.5883218      5   0.5</span></span>
<span><span class="co">#&gt; 11     B gene12   8.072284 0.4779155      5   0.5</span></span>
<span><span class="co">#&gt; 12     B gene13   9.938137 0.3418472      5   0.5</span></span>
<span><span class="co">#&gt; 13     B  gene2   8.273987 0.3147537      5   0.5</span></span>
<span><span class="co">#&gt; 14     B gene24   3.370467 0.7214700      5   0.5</span></span>
<span><span class="co">#&gt; 15     B gene21   5.789423 0.5279636      5   0.5</span></span>
<span><span class="co">#&gt; 16     B gene41   5.518354 0.6317005      5   0.5</span></span>
<span><span class="co">#&gt; 17     B gene74   3.226598 0.4940783      5   0.5</span></span>
<span><span class="co">#&gt; 18     B gene52   2.438579 0.5173769      5   0.5</span></span>
<span><span class="co">#&gt; 19     B gene95   6.288173 0.4462475      5   0.5</span></span>
<span><span class="co">#&gt; 20     B gene66   7.891588 0.5883218      5   0.5</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="recipe">
<code>recipe</code><a class="anchor" aria-label="anchor" href="#recipe"></a>
</h4>
<p><code>reclanc</code> also supports <code>tidymodels</code>
workflows:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://parsnip.tidymodels.org/reference/discrim_linear.html" class="external-link">discrim_linear</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://parsnip.tidymodels.org/reference/set_engine.html" class="external-link">set_engine</a></span><span class="op">(</span><span class="st">"clanc"</span>, active <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://generics.r-lib.org/reference/fit.html" class="external-link">fit</a></span><span class="op">(</span><span class="va">class</span> <span class="op">~</span> <span class="va">.</span>, data <span class="op">=</span> <span class="va">form_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; parsnip model object</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; &lt;clanc&gt; </span></span>
<span><span class="co">#&gt; $centroids</span></span>
<span><span class="co">#&gt;    class   gene expression pooled_sd active prior</span></span>
<span><span class="co">#&gt; 1      A gene12   7.514718 0.4779155      5   0.5</span></span>
<span><span class="co">#&gt; 2      A  gene2   8.744821 0.3147537      5   0.5</span></span>
<span><span class="co">#&gt; 3      A gene13   8.936462 0.3418472      5   0.5</span></span>
<span><span class="co">#&gt; 4      A gene21   6.584681 0.5279636      5   0.5</span></span>
<span><span class="co">#&gt; 5      A gene24   4.307301 0.7214700      5   0.5</span></span>
<span><span class="co">#&gt; 6      A gene74   4.028507 0.4940783      5   0.5</span></span>
<span><span class="co">#&gt; 7      A gene41   4.328516 0.6317005      5   0.5</span></span>
<span><span class="co">#&gt; 8      A gene95   6.873184 0.4462475      5   0.5</span></span>
<span><span class="co">#&gt; 9      A gene52   3.743798 0.5173769      5   0.5</span></span>
<span><span class="co">#&gt; 10     A gene66   7.008174 0.5883218      5   0.5</span></span>
<span><span class="co">#&gt; 11     B gene12   8.072284 0.4779155      5   0.5</span></span>
<span><span class="co">#&gt; 12     B gene13   9.938137 0.3418472      5   0.5</span></span>
<span><span class="co">#&gt; 13     B  gene2   8.273987 0.3147537      5   0.5</span></span>
<span><span class="co">#&gt; 14     B gene24   3.370467 0.7214700      5   0.5</span></span>
<span><span class="co">#&gt; 15     B gene21   5.789423 0.5279636      5   0.5</span></span>
<span><span class="co">#&gt; 16     B gene41   5.518354 0.6317005      5   0.5</span></span>
<span><span class="co">#&gt; 17     B gene74   3.226598 0.4940783      5   0.5</span></span>
<span><span class="co">#&gt; 18     B gene52   2.438579 0.5173769      5   0.5</span></span>
<span><span class="co">#&gt; 19     B gene95   6.288173 0.4462475      5   0.5</span></span>
<span><span class="co">#&gt; 20     B gene66   7.891588 0.5883218      5   0.5</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="tall-inputs">Tall inputs<a class="anchor" aria-label="anchor" href="#tall-inputs"></a>
</h3>
<p>Tall inputs require genes as rows and samples as columns</p>
<div class="section level4">
<h4 id="data-framematrix">
<code>data.frame</code>/<code>matrix</code><a class="anchor" aria-label="anchor" href="#data-framematrix"></a>
</h4>
<p>It is often convenient to supply a <code>data.frame</code>,
particularly after some data-munging has been done. Both
<code>data.frame</code> and <code>matrix</code> inputs require
expression with genes as column names and sample IDs as rownames, as
well as a factor vector of classes:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/clanc.html">clanc</a></span><span class="op">(</span></span>
<span>  <span class="va">synthetic_expression</span><span class="op">$</span><span class="va">expression</span>,</span>
<span>  classes <span class="op">=</span> <span class="va">synthetic_expression</span><span class="op">$</span><span class="va">classes</span>,</span>
<span>  active <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; &lt;clanc&gt; </span></span>
<span><span class="co">#&gt; $centroids</span></span>
<span><span class="co">#&gt;    class   gene expression pooled_sd active prior</span></span>
<span><span class="co">#&gt; 1      A gene12   7.514718 0.4779155      5   0.5</span></span>
<span><span class="co">#&gt; 2      A  gene2   8.744821 0.3147537      5   0.5</span></span>
<span><span class="co">#&gt; 3      A gene13   8.936462 0.3418472      5   0.5</span></span>
<span><span class="co">#&gt; 4      A gene21   6.584681 0.5279636      5   0.5</span></span>
<span><span class="co">#&gt; 5      A gene24   4.307301 0.7214700      5   0.5</span></span>
<span><span class="co">#&gt; 6      A gene74   4.028507 0.4940783      5   0.5</span></span>
<span><span class="co">#&gt; 7      A gene41   4.328516 0.6317005      5   0.5</span></span>
<span><span class="co">#&gt; 8      A gene95   6.873184 0.4462475      5   0.5</span></span>
<span><span class="co">#&gt; 9      A gene52   3.743798 0.5173769      5   0.5</span></span>
<span><span class="co">#&gt; 10     A gene66   7.008174 0.5883218      5   0.5</span></span>
<span><span class="co">#&gt; 11     B gene12   8.072284 0.4779155      5   0.5</span></span>
<span><span class="co">#&gt; 12     B gene13   9.938137 0.3418472      5   0.5</span></span>
<span><span class="co">#&gt; 13     B  gene2   8.273987 0.3147537      5   0.5</span></span>
<span><span class="co">#&gt; 14     B gene24   3.370467 0.7214700      5   0.5</span></span>
<span><span class="co">#&gt; 15     B gene21   5.789423 0.5279636      5   0.5</span></span>
<span><span class="co">#&gt; 16     B gene41   5.518354 0.6317005      5   0.5</span></span>
<span><span class="co">#&gt; 17     B gene74   3.226598 0.4940783      5   0.5</span></span>
<span><span class="co">#&gt; 18     B gene52   2.438579 0.5173769      5   0.5</span></span>
<span><span class="co">#&gt; 19     B gene95   6.288173 0.4462475      5   0.5</span></span>
<span><span class="co">#&gt; 20     B gene66   7.891588 0.5883218      5   0.5</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="summarizedexperiment">
<code>SummarizedExperiment</code><a class="anchor" aria-label="anchor" href="#summarizedexperiment"></a>
</h4>
<p>Some common formats for expression are
<code>SummarizedExperiment</code>s and <code>ExpressionSet</code>s:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span></span>
<span>  <span class="va">synthetic_expression</span><span class="op">$</span><span class="va">expression</span>,</span>
<span>  colData <span class="op">=</span> <span class="fu">DataFrame</span><span class="op">(</span>class <span class="op">=</span> <span class="va">synthetic_expression</span><span class="op">$</span><span class="va">classes</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">se</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 100 12 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): ''</span></span>
<span><span class="co">#&gt; rownames(100): gene1 gene2 ... gene99 gene100</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(12): sample1 sample2 ... sample11 sample12</span></span>
<span><span class="co">#&gt; colData names(1): class</span></span></code></pre>
<p>We can specify the name of the <code>colData</code>
(<code>pData</code> for <code>ExpressionSet</code>s) column that
contains the classes with the <code>classes</code> argument:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clanc.html">clanc</a></span><span class="op">(</span></span>
<span>  <span class="va">se</span>,</span>
<span>  classes <span class="op">=</span> <span class="st">"class"</span>,</span>
<span>  active <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  assay <span class="op">=</span> <span class="fl">1</span> <span class="co"># Index of assay - SummarizedExperiments only</span></span>
<span><span class="op">)</span></span>
<span><span class="va">fit</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; &lt;clanc&gt; </span></span>
<span><span class="co">#&gt; $centroids</span></span>
<span><span class="co">#&gt;    class    gene expression pooled_sd active prior</span></span>
<span><span class="co">#&gt; 1      A  gene11  2.2992343 1.2044848     20   0.5</span></span>
<span><span class="co">#&gt; 2      A   gene2  8.7448209 0.3147537     20   0.5</span></span>
<span><span class="co">#&gt; 3      A  gene13  8.9364621 0.3418472     20   0.5</span></span>
<span><span class="co">#&gt; 4      A  gene20  2.1925558 1.3104010     20   0.5</span></span>
<span><span class="co">#&gt; 5      A  gene10  4.9557850 0.8571716     20   0.5</span></span>
<span><span class="co">#&gt; 6      A  gene21  6.5846813 0.5279636     20   0.5</span></span>
<span><span class="co">#&gt; 7      A gene100  5.6455200 0.6175104     20   0.5</span></span>
<span><span class="co">#&gt; 8      A  gene22  6.1650079 0.4699756     20   0.5</span></span>
<span><span class="co">#&gt; 9      A  gene46  6.7344030 0.8233370     20   0.5</span></span>
<span><span class="co">#&gt; 10     A  gene24  4.3073008 0.7214700     20   0.5</span></span>
<span><span class="co">#&gt; 11     A  gene15  2.4254020 1.1910158     20   0.5</span></span>
<span><span class="co">#&gt; 12     A  gene25  5.0353875 0.7498139     20   0.5</span></span>
<span><span class="co">#&gt; 13     A  gene17  2.9424148 0.6628466     20   0.5</span></span>
<span><span class="co">#&gt; 14     A   gene4  4.3839026 0.7144711     20   0.5</span></span>
<span><span class="co">#&gt; 15     A  gene56  6.3441126 0.4078736     20   0.5</span></span>
<span><span class="co">#&gt; 16     A  gene41  4.3285163 0.6317005     20   0.5</span></span>
<span><span class="co">#&gt; 17     A  gene57  4.2237139 0.9531773     20   0.5</span></span>
<span><span class="co">#&gt; 18     A   gene7  5.5545202 0.7875124     20   0.5</span></span>
<span><span class="co">#&gt; 19     A  gene58  5.6162919 0.8161951     20   0.5</span></span>
<span><span class="co">#&gt; 20     A  gene12  7.5147181 0.4779155     20   0.5</span></span>
<span><span class="co">#&gt; 21     A   gene6  5.1534126 0.6194184     20   0.5</span></span>
<span><span class="co">#&gt; 22     A  gene51  6.6256136 0.7737520     20   0.5</span></span>
<span><span class="co">#&gt; 23     A  gene60  4.7434923 1.2945446     20   0.5</span></span>
<span><span class="co">#&gt; 24     A  gene52  3.7437977 0.5173769     20   0.5</span></span>
<span><span class="co">#&gt; 25     A  gene63  8.9293980 0.5635262     20   0.5</span></span>
<span><span class="co">#&gt; 26     A  gene53  4.3774614 0.8370528     20   0.5</span></span>
<span><span class="co">#&gt; 27     A  gene66  7.0081742 0.5883218     20   0.5</span></span>
<span><span class="co">#&gt; 28     A  gene83  3.6532038 0.8444393     20   0.5</span></span>
<span><span class="co">#&gt; 29     A  gene67  6.1384613 0.3677756     20   0.5</span></span>
<span><span class="co">#&gt; 30     A  gene85  5.2179679 0.5930857     20   0.5</span></span>
<span><span class="co">#&gt; 31     A  gene88  4.6008044 1.0603007     20   0.5</span></span>
<span><span class="co">#&gt; 32     A  gene70  1.3073340 1.1264747     20   0.5</span></span>
<span><span class="co">#&gt; 33     A  gene47  9.4528373 0.2030726     20   0.5</span></span>
<span><span class="co">#&gt; 34     A  gene90  0.9794695 1.3272423     20   0.5</span></span>
<span><span class="co">#&gt; 35     A  gene74  4.0285071 0.4940783     20   0.5</span></span>
<span><span class="co">#&gt; 36     A  gene94  7.7773183 0.5375914     20   0.5</span></span>
<span><span class="co">#&gt; 37     A  gene78  2.1763395 1.6805560     20   0.5</span></span>
<span><span class="co">#&gt; 38     A  gene95  6.8731844 0.4462475     20   0.5</span></span>
<span><span class="co">#&gt; 39     A  gene79  3.7138831 1.0587367     20   0.5</span></span>
<span><span class="co">#&gt; 40     A  gene98  4.5710407 0.6798799     20   0.5</span></span>
<span><span class="co">#&gt; 41     B  gene10  4.2378889 0.8571716     20   0.5</span></span>
<span><span class="co">#&gt; 42     B   gene2  8.2739866 0.3147537     20   0.5</span></span>
<span><span class="co">#&gt; 43     B gene100  5.0435040 0.6175104     20   0.5</span></span>
<span><span class="co">#&gt; 44     B  gene20  3.4781598 1.3104010     20   0.5</span></span>
<span><span class="co">#&gt; 45     B  gene46  7.0200767 0.8233370     20   0.5</span></span>
<span><span class="co">#&gt; 46     B  gene11  1.2780748 1.2044848     20   0.5</span></span>
<span><span class="co">#&gt; 47     B  gene12  8.0722841 0.4779155     20   0.5</span></span>
<span><span class="co">#&gt; 48     B  gene22  6.4609169 0.4699756     20   0.5</span></span>
<span><span class="co">#&gt; 49     B  gene51  5.8920005 0.7737520     20   0.5</span></span>
<span><span class="co">#&gt; 50     B  gene13  9.9381374 0.3418472     20   0.5</span></span>
<span><span class="co">#&gt; 51     B  gene15  1.6008569 1.1910158     20   0.5</span></span>
<span><span class="co">#&gt; 52     B  gene25  4.5015558 0.7498139     20   0.5</span></span>
<span><span class="co">#&gt; 53     B  gene17  2.5005839 0.6628466     20   0.5</span></span>
<span><span class="co">#&gt; 54     B   gene4  4.9225469 0.7144711     20   0.5</span></span>
<span><span class="co">#&gt; 55     B  gene56  6.1067832 0.4078736     20   0.5</span></span>
<span><span class="co">#&gt; 56     B  gene41  5.5183538 0.6317005     20   0.5</span></span>
<span><span class="co">#&gt; 57     B  gene57  3.1175271 0.9531773     20   0.5</span></span>
<span><span class="co">#&gt; 58     B   gene7  5.3367575 0.7875124     20   0.5</span></span>
<span><span class="co">#&gt; 59     B  gene21  5.7894231 0.5279636     20   0.5</span></span>
<span><span class="co">#&gt; 60     B  gene47  9.5903798 0.2030726     20   0.5</span></span>
<span><span class="co">#&gt; 61     B   gene6  4.4655748 0.6194184     20   0.5</span></span>
<span><span class="co">#&gt; 62     B  gene74  3.2265977 0.4940783     20   0.5</span></span>
<span><span class="co">#&gt; 63     B  gene24  3.3704670 0.7214700     20   0.5</span></span>
<span><span class="co">#&gt; 64     B  gene52  2.4385792 0.5173769     20   0.5</span></span>
<span><span class="co">#&gt; 65     B  gene63  8.3234317 0.5635262     20   0.5</span></span>
<span><span class="co">#&gt; 66     B  gene53  3.8479638 0.8370528     20   0.5</span></span>
<span><span class="co">#&gt; 67     B  gene66  7.8915875 0.5883218     20   0.5</span></span>
<span><span class="co">#&gt; 68     B  gene83  4.2757218 0.8444393     20   0.5</span></span>
<span><span class="co">#&gt; 69     B  gene67  6.0190764 0.3677756     20   0.5</span></span>
<span><span class="co">#&gt; 70     B  gene85  5.8877225 0.5930857     20   0.5</span></span>
<span><span class="co">#&gt; 71     B  gene79  4.1894417 1.0587367     20   0.5</span></span>
<span><span class="co">#&gt; 72     B  gene58  4.7194615 0.8161951     20   0.5</span></span>
<span><span class="co">#&gt; 73     B  gene88  5.5945405 1.0603007     20   0.5</span></span>
<span><span class="co">#&gt; 74     B  gene70  1.5987845 1.1264747     20   0.5</span></span>
<span><span class="co">#&gt; 75     B  gene90  1.4036889 1.3272423     20   0.5</span></span>
<span><span class="co">#&gt; 76     B  gene60  5.2336968 1.2945446     20   0.5</span></span>
<span><span class="co">#&gt; 77     B  gene78  1.6625207 1.6805560     20   0.5</span></span>
<span><span class="co">#&gt; 78     B  gene95  6.2881728 0.4462475     20   0.5</span></span>
<span><span class="co">#&gt; 79     B  gene98  4.1346296 0.6798799     20   0.5</span></span>
<span><span class="co">#&gt; 80     B  gene94  8.4222554 0.5375914     20   0.5</span></span></code></pre>
</div>
</div>
</div>
<div class="section level2">
<h2 id="predicting">Predicting<a class="anchor" aria-label="anchor" href="#predicting"></a>
</h2>
<p>The fit can then be used to predict classes of new samples from new
data. The new data can come in the form of a <code>matrix</code>,
<code>data.frame</code>, <code>SummarizedExperiment</code>, or
<code>ExpressionSet</code>, with the same expected input</p>
<p>Using <code>type = "class"</code> will predict classes using the
metric provided by Alan Dabney in the original ClaNC paper.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">fit</span>, new_data <span class="op">=</span> <span class="va">se</span>, type <span class="op">=</span> <span class="st">"class"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; 40/40 (100%) genes in centroids found in data</span></span></code></pre>
<pre><code><span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 1</span></span></span>
<span><span class="co">#&gt;    .pred_class</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> A          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> A          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> A          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> A          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> A          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> A          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> B          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> B          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> B          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> B          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> B          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> B</span></span></code></pre>
<p>However, particularly if comparing across datasets that may have been
transformed differently, it may be more accurate to use a correlation
based metric:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">fit</span>, new_data <span class="op">=</span> <span class="va">se</span>, type <span class="op">=</span> <span class="st">"numeric"</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; 40/40 (100%) genes in centroids found in data</span></span></code></pre>
<pre><code><span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 2</span></span></span>
<span><span class="co">#&gt;    .pred_A .pred_B</span></span>
<span><span class="co">#&gt;      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>   0.901   0.811</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>   0.929   0.849</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>   0.932   0.840</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>   0.912   0.829</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>   0.862   0.770</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>   0.932   0.869</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>   0.776   0.904</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>   0.824   0.931</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>   0.828   0.924</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>   0.855   0.946</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>   0.805   0.915</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span>   0.750   0.869</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Kai Aragaki, Alan Dabney.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
